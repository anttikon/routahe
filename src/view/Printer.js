import {getColorByMode, getEmojiByMode, formatTime, formatDuration} from './viewUtils'
import {cyan, gray} from 'chalk'

const dateTimeFormat = 'H:mm DD.MM.YYYY'

class Printer {

  addDate(date) {
    if (!date) {
      return this
    }
    return this._addLine(formatTime(date))
  }

  addDuration(duration) {
    if (!duration) {
      return this
    }
    return this._addLine(formatDuration(duration))
  }

  addMode(mode) {
    if (!mode) {
      return this
    }
    const color = getColorByMode(mode)
    return this._addLine(`${getEmojiByMode(mode)} ${color(mode)}`)
  }

  addArrive(arrive) {
    if (!arrive) {
      return this
    }
    this.addString('arrive ', gray)
    return this.addString(arrive.format(dateTimeFormat), cyan)
  }

  addDeparture(departure) {
    if (!departure) {
      return this
    }
    this.addString('departure ', gray)
    return this.addString(departure.format(dateTimeFormat), cyan)
  }

  addString(str, color) {
    if (!str) {
      return this
    }

    return color ? this._addLine(color(str)) : this._addLine(str)
  }

  _addLine(line) {
    if (!this.lines) {
      this.lines = []
    }
    this.lines.push(line)
    return this
  }

  print() {
    this.lines ? console.log(this.lines.join('')) : null
  }
}
export default Printer