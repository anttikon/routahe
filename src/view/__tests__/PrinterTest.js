import Printer from '../Printer'
import moment from 'moment'
import {cyan} from 'chalk'
import sinon from 'sinon'

describe('Printer', function() {

  beforeEach(() => sinon.spy(console, 'log'))
  afterEach(() => console.log.restore())

  it('should be able to print line', () => {
    new Printer()
      .addDate(new Date(1481728405559))
      .addDuration(47285)
      .addMode('SUBWAY')
      .addString(' ')
      .addArrive(moment(1481728405559))
      .addDeparture(moment(1481728405559))
      .addString(' | ', cyan)
      .print()

    expect(console.log.getCall(0).args).toMatchSnapshot()
  })

  it('should return empty stuff if parameters empty ok', () => {
    new Printer()
      .addDate()
      .addDuration()
      .addMode()
      .addString()
      .addArrive()
      .addDeparture()
      .addString(null, cyan)
      .addString('a')
      .print()

    expect(console.log.getCall(0).args).toMatchSnapshot()
  })
})