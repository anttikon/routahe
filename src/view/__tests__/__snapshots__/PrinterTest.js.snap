exports[`Printer should be able to print line 1`] = `
Array [
  "17:1313h 08minðŸš‡ [33mSUBWAY[39m [90marrive [39m[36m17:13 14.12.2016[39m[90mdeparture [39m[36m17:13 14.12.2016[39m[36m | [39m",
]
`;

exports[`Printer should return empty stuff if parameters empty ok 1`] = `
Array [
  "a",
]
`;
